staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -21,39 +21,246 @@ import (
+	}
+
+	DisallowRoot(pod, containerName, podutils.Containers|podutils.EphemeralContainers)
+	DropCapabilities(pod, containerName, podutils.Containers|podutils.EphemeralContainers)
+
+	switch style {
+	case node:
+		ClearSecurityContext(pod, containerName, podutils.Containers)
+
+	case podCopy:
+		ShareProcessNamespace(pod)
+
+	case ephemeral:
+		// no additional modifications needed
+	}
+
+	return nil
+}
+
+// RemoveLabelsAndProbes removes labels from the pod and remove probes
+// from all containers of the pod.
+func RemoveLabelsAndProbes(p *corev1.Pod) {
+	p.Labels = nil
+	for i := range p.Spec.Containers {
+		p.Spec.Containers[i].LivenessProbe = nil
+		p.Spec.Containers[i].ReadinessProbe = nil
+	}
+}
+
+// MountRootPartition mounts the host's root path at "/host" in the container.
+func MountRootPartition(p *corev1.Pod, containerName string) {
+	const volumeName = "host-root"
+	p.Spec.Volumes = append(p.Spec.Volumes, corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: "/"},
+		},
+	})
+	podutils.VisitContainers(&p.Spec, podutils.Containers, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+			MountPath: "/host",
+			Name:      volumeName,
+		})
+		return false
+	})
+}
+
+// UseHostNamespaces configures the pod to use the host's network, PID, and IPC
+// namespaces.
+func UseHostNamespaces(p *corev1.Pod) {
+	p.Spec.HostNetwork = true
+	p.Spec.HostPID = true
+	p.Spec.HostIPC = true
+}
+
+// ShareProcessNamespace configures all containers in the pod to share the
+// process namespace.
+func ShareProcessNamespace(p *corev1.Pod) {
+	p.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+}
+
+// ClearSecurityContext clears the security context for the container.
+func ClearSecurityContext(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.SecurityContext = nil
+		return false
+	})
+}
+
+// DisallowRoot configures the container to run as a non-root user.
+func DisallowRoot(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.SecurityContext = &corev1.SecurityContext{
+			RunAsNonRoot: pointer.BoolPtr(true),
+		}
+		return false
+	})
+}
+
+// DropCapabilities drops all Capabilities for the container
+func DropCapabilities(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		if c.SecurityContext == nil {
```

Comment:
It's a little difficult to see that these all do the same thing, and I think we probably want to overwrite everything anyway if the goal is to drop them all. What about this?

```
if c.SecurityContext == nil {
  c.SecurityContext = &corev1.SecurityContext{}
}
c.SecurityContext.Capabilities = &corev1.Capabilities{
  Drop: []corev1.Capability{"ALL"},
}
return false
```

Reply:
good idea - done in f337511f064d7e28fa0bf7ab4b27a3af17d178d9